### ISSUE TITLE
Pylint crash with ``StopIteration`` when used with ``--limit-inference-results 0``

### ISSUE THREAD

**Issue by MikhailZex** [PyCQA/pylint#4631]
### Steps to reproduce

1. ```
    pylint --fail-under 9 --limit-inference-results 0 ./spotdl
    ```
    on any file/files that import from modules that haven't been installed

2. Discovered on GitHub actions run that you can check [here](https://github.com/spotDL/spotdlRedesign/pull/8/checks?check_run_id=2948836676).

### pylint --version output

Result of `pylint --version` output:

```
2.9.0
```


### Notes

1. Sorry for not strictly adhering to the template. All the files, previous checks and current check outputs can be seen on GitHub itself.

2. This is an issue I found only with 2.9.0
<!--
Additional dependencies:
```
pandas==0.23.2
marshmallow==3.10.0
...
```
-->

**Comment by Pierre-Sassoulas** [comment]
Thanks for opening the issue, I can reproduce this problem.

**Comment by Pierre-Sassoulas** [comment]
This should be fixed if upgrading to astroid 2.6.2 manually, and will be fixed in pylint 2.9.1.

**Comment by MikhailZex** [comment]
Thanks for the timely help.
